src/classes/CRLP_Query_SEL.cls

```diff
@@ -352,7 +352,7 @@
                     // If the value doesn't match though, the countOfRuleUsage map value is decremented to indicate
                     // that the Rule for the field isn't used consistently across all Filter Groups
                     if (!targetFieldAndConstant.containsKey(fieldName)) {
-                        targetFieldAndConstant.put(fieldName, rule.Constant__c);
+                        targetFieldAndConstant.put(fieldName, rule.Constant__c == null ? '' : rule.Constant__c);
```

Comment:
This is a good check for null, however if an empty string is used here, then the `if` below might not match (null != ""). I suggest using something like the following above the `if` block to reference the constant value:
`String constantVal = (rule.Constant__c == null ? '' : rule.Constant__c);`